# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/ServiceWorkerOp.cpp
# Commit: 8fec60d0c499
# Full Hash: 8fec60d0c499fdf429f4d11da6bea0c65aecb1d4
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2022-02-27 09:40:54
# Regressor Bug: 1754365
# File Overlap Count: 1
# Description:
#   Bug 1754365 - Create IPC for transferring NavigationPreload's PerformanceTimingData. r=dom-worker-reviewers,jesup
#   
#   Depends on D138250
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138251
# ==============================================================================

diff -r fb0ac558803d -r 8fec60d0c499 dom/serviceworkers/ServiceWorkerOp.cpp
--- a/dom/serviceworkers/ServiceWorkerOp.cpp	Sun Feb 27 00:26:21 2022 +0000
+++ b/dom/serviceworkers/ServiceWorkerOp.cpp	Sun Feb 27 00:26:22 2022 +0000
@@ -1682,10 +1682,16 @@
         ->Then(
             GetCurrentSerialEventTarget(), __func__,
             [self, globalObjectAsSupports = std::move(globalObjectAsSupports)](
-                SafeRefPtr<InternalResponse> aInternalResponse) {
-              self->mPreloadResponse->MaybeResolve(
-                  MakeRefPtr<Response>(globalObjectAsSupports,
-                                       std::move(aInternalResponse), nullptr));
+                FetchEventPreloadResponseArgs&& aArgs) {
+              SafeRefPtr<InternalResponse> preloadResponse;
+              IPCPerformanceTimingData timingData;
+              nsString initiatorType;
+              nsString entryName;
+              Tie(preloadResponse, timingData, initiatorType, entryName) =
+                  std::move(aArgs);
+
+              self->mPreloadResponse->MaybeResolve(MakeRefPtr<Response>(
+                  globalObjectAsSupports, std::move(preloadResponse), nullptr));
               self->mPreloadResponsePromiseRequestHolder.Complete();
             },
             [self](int) {